modules/xfem/include/userobjects/LevelSetCutUserObject.h

```diff
@@ -0,0 +1,48 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef LEVELSETCUTUSEROBJECT_H
+#define LEVELSETCUTUSEROBJECT_H
+
+#include "GeometricCutUserObject.h"
+
+class LevelSetCutUserObject;
+
+template <>
+InputParameters validParams<LevelSetCutUserObject>();
+
+class LevelSetCutUserObject : public GeometricCutUserObject
+{
+public:
+  LevelSetCutUserObject(const InputParameters & parameters);
+
+  virtual bool cutElementByGeometry(const Elem * elem,
+                                    std::vector<Xfem::CutEdge> & cut_edges,
+                                    std::vector<Xfem::CutNode> & cut_nodes,
+                                    Real time) const override;
+  virtual bool cutElementByGeometry(const Elem * elem,
+                                    std::vector<Xfem::CutFace> & cut_faces,
+                                    Real time) const override;
+
+  virtual bool cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_edges,
+                                     std::vector<Xfem::CutEdge> & cut_edges,
+                                     Real time) const override;
+  virtual bool cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_faces,
+                                     std::vector<Xfem::CutFace> & cut_faces,
+                                     Real time) const override;
+
+  virtual const std::vector<Point>
+  getCrackFrontPoints(unsigned int num_crack_front_points) const override;
```

Comment:
Add documentation for all of these five methods please, including why the first four are different variations on overloading the method

Reply:
The first four functions are defined in the base class ```GeometricCutUserObject``` and might be better to document them there.